### ISSUE TITLE
Output broken when compiling foundation 5.4.7 with output style compressed

### ISSUE THREAD

**Issue by Sgoettschkes** [sass/libsass#632]
I am facing a problem when compiling foundation 5.4.7 using grunt-sass 0.16.1 which uses node-sass 1.0.3 which uses libsass (I am not sure which version). 

Compiling the [foundation.scss](https://github.com/zurb/foundation/blob/master/scss/foundation.scss) file with the output style compressed causes some stuff to be not included in the output file. I have created a [gist](https://gist.github.com/Sgoettschkes/67076918ccdc4934db7c) with [sass.css.min](https://gist.githubusercontent.com/Sgoettschkes/67076918ccdc4934db7c/raw/00b9354ce8329bbc83a84fe246926c540d9bff32/sass.min.css) and [libsass.css.min](https://gist.githubusercontent.com/Sgoettschkes/67076918ccdc4934db7c/raw/5bdc7ef585de1b0120bbc7f1143271c661d2483e/libsass.min.css). sass.min.css was compiled with

    sass -t compressed bower_components/foundation/scss/foundation.scss sass.min.css

and libsass.min.css was compiled with grunt, using this config:

    sass: {
        dist: {
            options: {
                outputStyle: 'compressed',
                sourceMap: true
            },
            files: {
                'libsass.min.css': 'bower_components/foundation/scss/foundation.scss'
            }
        }
    }

As you can see, the output is different. I am not sure what is stripped, but when using the libsass.min.css on my website, I can see that css classes `small-1` to `small-12`, `medium-1` to `medium-12` and `large-1` to `large-12` are missing.

Using another output style (compact or expanded) doesn't cause this error. 

I am not sure what is happening and how to debug any further. If you need any more information I am willing to provide anything needed. If this issue was already reported or is caused by something else, I am more than happy to report it to another project as well.

**Comment by STRML** [comment]
I am seeing similar issues in my application where classes go missing. It appears to be related to media queries; I am seeing lots of empty media queries in the output, and the content that is actually in the queries does not appear to be printed.

For example:

```
.sideWrap {
  .showHide {
    $padding: 20px - 2px;
    height: pem(10);
    position: relative;
    margin-bottom: 3px;
    .fa {
      position: absolute;
      top: -2px;
      opacity: 0.4;
    }
    .one {
      left: $padding - 4px;
    }
    .two {
      right: $padding;
    }
  }
  .measuringWrapper {
    padding-bottom: 10px;
  }
  @media (max-width: $screen-xs-max) {
    .showHide {
      display: none;
    }
  }
}
.side-nav .panel-collapse li.separator {
  border-bottom: 1px solid #aaa;
  font-weight: 700;
  margin: 5px 10px 5px 30px;
}
```

Compiles to:

```
@media (min-width: 996px) {}@media (max-width: 767px) {}@media (max-width: 480px) {}.sideWrap .showHide {
    height:0.7692307692em;
    position:relative;
    margin-bottom:3px
}
.sideWrap .showHide .fa {
    position:absolute;
    top:-2px;
    opacity:0.4
}
.sideWrap .showHide .one {
    left: 14px
}
.sideWrap .showHide .two {
    right: 18px
}
.sideWrap .measuringWrapper {
    padding-bottom: 10px
}
@media (max-width: 767px) {}.side-nav .panel-collapse li.separator {
    border-bottom:1px solid #aaa;
    font-weight:700;
    margin:5px 10px 5px 30px
}
```

This is a really nasty bug because it is only showing up in `compressed` mode, which means I generally don't catch it until it hits staging.

**Comment by xzyfer** [comment]
This should be fixed in 3.0.2
